### DISCUSSION TITLE
Do we have Cumulative Volume Delta .?

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#1252] [Ideas]
Hi @DaveSkender,
we have an indicator Cumulative Volume Delta in the trading view attached some reference image as below

Do we have the same indicator in our library I searched in the list but didn't found with same name 

Also, can we add the same to our library if it's not there?

![image](https://github.com/user-attachments/assets/e2ebe4c6-880d-4045-af38-7e938188ba10)

![image](https://github.com/user-attachments/assets/8946dfdc-06d4-43f5-855b-b2c9a176fa84)

**Comment by DaveSkender** [comment]
Not directly, no.  Though am trying to think through some currently available chaining options that would do this.  ROC of directional volume might get you partly to this one.  Also looks a bit like Elder-Ray, but can’t make the connection.

**Comment by JGronholz** [comment]
When you subscribe to the trades feed, you can simply accumulate each executed trade's volume into a total, adding buys, and subtracting sells. This is the real, one-and-only, CVD. Since it is real-time, it doesn't matter how you slice it, after the fact.

If your app integrates it's own candles from the trades feed, your candles would include BuyVolume, SellVolume, and the total/net Volume. Having already witnessed those trades, integrated them into candles, and persisted them, you could reconstruct the CVD as precisely as you persisted the trades.

The problem with estimating CVD from candles that don't include that derived information (like Buy/Sell volume), is that you don't know how much of Volume is which. With TV, you get Volume, Open, and Close. If Close is higher than Open, you give it to the buyers, and vice-versa.

The problem is that there is not enough precision in a series of X-hour candles to re-construct the X-hour CVD (precisely).
To reconstruct a more-accurate X-hour CVD, you need to use candles from a timeframe shorter than X.

For example, one 1-hour candle with wicks in both directions could be:
-one 15-minute candle from Open to Low (Volume-)
-one 15-minute candle from Low back to Open (Volume+)
-one 15-minute candle from Open to High (Volume+)
-one 15-minute candle from High to Close (Volume-)

This still accumulates error by choosing a side based solely on (Close > Open), but this method would have (1-hour / 15-minute =) 4X granularity over using 1-hour data alone.

var CVD = lowestTimeframeCandles.Sum(candle => candle.Close >= candle.Open ?? candle.Volume : -candle.Volume)

Anchor the CVD:
var CVD = lowestTimeframeCandles.Where(candle => candle.Timestamp >= AnchorTimestamp).Sum(...)

So, the problems with this "indicator":
-it requires quotes from a different timeframe than the results (doable)
-it is just a single Linq statement

If your app isn't just for historical viewing, subscribe to the trades feed and Sum the trades directly, based on Side. Be clever/realistic about persisting the trades feed verbatim.

**Reply by JGronholz** [reply]
For the time-series, CVDResult.CVD would just be the running total, and CVDResults.Last().CVD should be equivalent to my Linq statement above.
